Issue237

erosquesada opened the pull request:
Now the function takes into account the differences in low-hierarchies mentioned [here](https://vocab.ices.dk/?ref=1598). 
This allow to process data belonging to low-hierarchies having no FMid, and hence allow breaking the code in functions relying on this basis (e.g. those in RDBESvisualise package). The code was written in dplyr, then translated in data.table. 
Please @nmprista or @Kasia-MIR to check ahead of merging. @Kasia-MIR, once this is merged we can proceed.

erosquesada committed:
With dplyr code

erosquesada committed:
Fixed with data.table code, to check

erosquesada committed:
Fixed with data.table code, to check on #237

erosquesada committed:
replacing with x the object

Kasia-MIR committed:
required library

Kasia-MIR commented:
first check: 
`
# RDBESdataObject
library(RDBEScore)
library(data.table)
H1 <- createRDBESDataObject(input = "./TableFormat141025/All_H/2025_10_14_125156.zip", Hierarchy = 1)

# use createTableOfRDBESIds
H1_ids <- createTableOfRDBESIds(H1)

#check 1
#BVid without BVfishid
case1 <-H1_ids[is.na(H1_ids$BVid) & !is.na(H1_ids$BVfishId),] 
if(nrow(case1)!=0){
  stop("Error: BVfishId without BVid")
}
case2 <-H1_ids[!is.na(H1_ids$BVid) & is.na(H1_ids$BVfishId),] 
if(nrow(case2)!=0){
  stop("Error: BVid without BVfishId")
}
`

Kasia-MIR committed:
fix bug with BVfishId

passed the test: https://github.com/ices-tools-dev/RDBEScore/pull/246#issuecomment-3406981773

Kasia-MIR reviewed (approved):
Fixed bug with BVfishId.
